test/integration_test.go

```diff
@@ -684,6 +684,20 @@ func (ts *IntegrationTestSuite) TestContextPropagator() {
 	}, propagatedValues)
 }
 
+const CronWorkflowID = "test-cron"
+
+func (ts *IntegrationTestSuite) TestFailurePropagation() {
+	var expected int
+	err := ts.executeWorkflow(CronWorkflowID, ts.workflows.CronWorkflow, &expected)
+	// Workflow finishes itself by cancelling
```

Comment:
It is kind of a mess the way it is implemented. But the whole idea of cancellation is that it comes from outside. The workflow finishing itself by canceling is a misnomer.

Reply:
Sure, makes sense. Is there a better way I should do this, or just clarifying the comment?